streaming/base/dataset.py

```diff
@@ -151,11 +151,25 @@ def __init__(self,
         self.download_timeout = download_timeout
         self.validate_hash = validate_hash or None
 
+        # Initialize the World context.
+        #
+        # Beware: This information is for the per-rank process. DataLoader worker processes may see
+        # different values for these fields. We are saving the rank World here because we cannot
+        # instantiate a World inside the StreamingDataset destructor.
+        self._rank_world = world = World()
+
         if self.download_retry < 0:
             raise ValueError('Parameter ``download_retry`` must be non-negative')
         if self.download_timeout < 0:
             raise ValueError(
                 'Parameter ``download_timeout`` (in seconds) must be greater than zero')
+        local_split_path = os.path.join(self.local, self.split)
+        # Check if the directory exist and also not empty
+        if os.path.exists(local_split_path) and os.listdir(local_split_path):
+            raise ValueError(''.join([
+                f'Directory {local_split_path} already exist and not empty.',
```

Comment:
ditto

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/149/commits/9294569eaca43c9abafa10541597c735c5f96104.